# Profitable App Profiles for the App Store and Google Play Markets
For this project, we'll pretend we're working as data analysts for a company that builds Android and iOS mobile apps. We make our apps available on Google Play and the App Store.

We only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means our revenue for any given app is mostly influenced by the number of users who use our app — the more users that see and engage with the ads, the better. Our goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

In [1]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n')

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [2]:
from csv import reader

ios = list(reader(open('AppleStore.csv', encoding='utf8')))
ios_header = ios[0]
ios_data = ios[1: ]

android = list(reader(open('googleplaystore.csv', encoding='utf8')))
android_header = android[0]
android_body = android[1: ]

In [3]:
explore_data(ios, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns: 16


In [4]:
explore_data(android, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns: 13


In [5]:
# delete rows with missing data
del android[10473]

In [6]:
len(android)

10841

The Google Play data set has duplicate entries so we need to delete the duplicates first before analysis

In [7]:
unique_set = []
duplicate_set = []

for app in android[1:]:
    name = app[0]
    if name in unique_set:
        duplicate_set.append(name)
    else:
        unique_set.append(name)

In [8]:
print(len(android[1:]))
print(len(duplicate_set))
print(len(unique_set))

10840
1181
9659


As you can see their are 1181 duplicates entries in the Google play data set and we should delete them so it doesn't affect our analysis but which entry we should delete? should it be random? or should it have some criteria?!
lets check one of the apps that has duplicates and see...

In [9]:
for app in android:
    if app[0] == duplicate_set[5]:
        print(app)

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [10]:
for app in android:
    if app[0] == duplicate_set[40]:
        print(app)

['Hangouts', 'COMMUNICATION', '4.0', '3419249', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', 'July 21, 2018', 'Varies with device', 'Varies with device']
['Hangouts', 'COMMUNICATION', '4.0', '3419433', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', 'July 21, 2018', 'Varies with device', 'Varies with device']
['Hangouts', 'COMMUNICATION', '4.0', '3419513', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', 'July 21, 2018', 'Varies with device', 'Varies with device']
['Hangouts', 'COMMUNICATION', '4.0', '3419464', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', 'July 21, 2018', 'Varies with device', 'Varies with device']


After discovering some of the duplicates entries we can see that some duplicates have the same exact data and some differs on the number of 'reviews'
for the first case we can delete any entry randomly
But for the second case it makes sense to take the entry with the largest number of reviews
because the different numbers show the data was collected at different times so the largest number will mean the latest data

### To remove the duplicates, we will:

- Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app.

- Use the information stored in the dictionary and create a new data set, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews).

In [11]:
reviews_max = {}
for app in android[1: ]:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max:
        if float(reviews_max[name][3]) < n_reviews:
            reviews_max[name][3] = app[3]
    else:
        reviews_max[name] = app


print(len(reviews_max))

9659


In [12]:
print(reviews_max['Hangouts'][3])
print(reviews_max['Instagram'][3])

3419513
66577446


In [13]:
android_clean = []
already_added = []
for app in android[1: ]:
    name = app[0]
    n_reviews = float(app[3])
    if name not in already_added and n_reviews == float(reviews_max[name][3]):
        android_clean.append(app)
        already_added.append(name)


In [14]:
print(len(android_clean))

9659


## Removing Non English apps
Remember we use English for the apps we develop at our company, and we'd like to analyze only the apps that are directed toward an English-speaking audience. However, if we explore the data long enough, we'll find that both data sets have apps with names that suggest they are not directed toward an English-speaking audience

In [21]:
print(ios[2746][1])

大众点评-发现品质生活


We're not interested in keeping these apps, so we'll remove them. One way to go about this is to remove each app with a name containing a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;), and other symbols (+, *, /).

Behind the scenes, each character we use in a string has a corresponding number associated with it. For instance, the corresponding number for character 'a' is 97, character 'A' is 65, and character '爱' is 29,233. We can get the corresponding number of each character using the ord() built-in function

In [22]:
print(ord('A'))
print(ord('z'))
print(ord('爱'))

65
122
29233


The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII system. 
Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.

If an app name contains a character that is greater than 127, then it probably means that the app has a non-English name.

In [25]:
def isEnglish(word):
    for letter in word:
        if ord(letter) > 127:
            return False
    return True

In [30]:
print(isEnglish('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(isEnglish('Docs To Go™ Free Office Suite'))
print(isEnglish('Twitter'))
print(isEnglish('Instachat 😜'))

False
False
True
False


On the previous screen, we wrote a function that detects non-English app names, but we saw that the function couldn't correctly identify certain English app names like 'Docs To Go™ Free Office Suite' and 'Instachat 😜'. This is because emojis and characters like ™ fall outside the ASCII range and have corresponding numbers over 127.


If we're going to use the function we've created, we'll lose useful data since many English apps will be incorrectly labeled as non-English. To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range. This means all English apps with up to three emoji or other special characters will still be labeled as English. Our filter function is still not perfect, but it should be fairly effective.

In [31]:
def isEnglishOptimized(word):
    counter = 0
    for letter in word:
        if ord(letter) > 127:
            counter += 1
            if counter > 3:
                return False
    return True

In [32]:
print(isEnglishOptimized('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(isEnglishOptimized('Docs To Go™ Free Office Suite'))
print(isEnglishOptimized('Twitter'))
print(isEnglishOptimized('Instachat 😜'))

False
True
True
True


Much Better!
Now Lets delete the non-English apps from IOS and Google Play

In [33]:
android_english = []

for app in android_clean:
    name = app[0]
    if isEnglishOptimized(name):
        android_english.append(app)

print(len(android_english))

9614


In [42]:
ios_english = []

for app in ios[1:]:
    name = app[1]
    if isEnglishOptimized(name):
        ios_english.append(app)

print(len(ios_english))

6183


## Isolating the free apps
As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data sets contain both free and non-free apps; we'll need to isolate only the free apps for our analysis.

In [40]:
android_free = []

for app in android_english:
    a_type = app[6]
    if a_type == 'Free':
        android_free.append(app)

print(len(android_free))

8861


In [43]:
ios_free = []

for app in ios_english:
    price = float(app[4])
    if price == 0.0:
        ios_free.append(app)

print(len(ios_free))

3222


our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:
1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

## Most Common Apps By Genre
Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.

In [44]:
explore_data(android_free, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '974', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 8861
Number of columns: 13


After exploring the android data set, We can use the columns (Category, Genres) to generate frequency tables to find out what are the most common genres in Google Play market

In [45]:
explore_data(ios_free, 0, 5, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 3222
Number of columns: 16


After exploring the IOS data set, We can use the columns (prime_genre) to generate frequency tables to find out what are the most common genres in IOS market

We'll build two functions we can use to analyze the frequency tables:

- One function to generate frequency tables that show percentages
- Another function we can use to display the percentages in a descending order

dictionaries don't have order, and it will be very difficult to analyze the frequency tables. We'll need to build a second function which can help us display the entries in the frequency table in a descending order.

In [52]:
def freq_table(dataset, index):
    table = {}
    for row in dataset:
        column = row[index]
        if column in table:
            table[column] += 1
        else:
            table[column] = 1
            
    dataset_length = len(dataset)
    for row in table:
        table[row] = table[row] / dataset_length * 100
    
    return table

In [54]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [55]:
display_table(android_free, 1)

FAMILY : 18.440356618891773
GAME : 9.874731971560772
TOOLS : 8.441485159688522
BUSINESS : 4.593161042771697
LIFESTYLE : 3.9047511567543167
PRODUCTIVITY : 3.8934657487868187
FINANCE : 3.7016138133393524
MEDICAL : 3.521047285859384
SPORTS : 3.3969077982169056
PERSONALIZATION : 3.317909942444419
COMMUNICATION : 3.2389120866719328
HEALTH_AND_FITNESS : 3.080916375126961
PHOTOGRAPHY : 2.9454914795169844
NEWS_AND_MAGAZINES : 2.7987811759395105
SOCIAL : 2.663356280329534
TRAVEL_AND_LOCAL : 2.3360794492720913
SHOPPING : 2.245796185532107
BOOKS_AND_REFERENCE : 2.144227513824625
DATING : 1.8620923146371742
VIDEO_PLAYERS : 1.783094458864688
MAPS_AND_NAVIGATION : 1.3993905879697552
EDUCATION : 1.2865365082947748
FOOD_AND_DRINK : 1.2413948764247829
ENTERTAINMENT : 1.1285407967498025
LIBRARIES_AND_DEMO : 0.9366888613023362
AUTO_AND_VEHICLES : 0.9254034533348381
HOUSE_AND_HOME : 0.8351201895948538
WEATHER : 0.8012639656923597
EVENTS : 0.7109807019523756
ART_AND_DESIGN : 0.6771244780498815
PARENTING : 

In [56]:
display_table(android_free, 9)

Tools : 8.430199751721025
Entertainment : 6.071549486513937
Education : 5.349283376594064
Business : 4.593161042771697
Productivity : 3.8934657487868187
Lifestyle : 3.8934657487868187
Finance : 3.7016138133393524
Medical : 3.521047285859384
Sports : 3.464620246021894
Personalization : 3.317909942444419
Communication : 3.2389120866719328
Action : 3.103487191061957
Health & Fitness : 3.080916375126961
Photography : 2.9454914795169844
News & Magazines : 2.7987811759395105
Social : 2.663356280329534
Travel & Local : 2.3247940413045933
Shopping : 2.245796185532107
Books & Reference : 2.144227513824625
Simulation : 2.0426588421171425
Dating : 1.8620923146371742
Arcade : 1.8508069066696762
Video Players & Editors : 1.7718090508971898
Casual : 1.760523642929692
Maps & Navigation : 1.3993905879697552
Food & Drink : 1.2413948764247829
Puzzle : 1.1285407967498025
Racing : 0.9931159011398263
Role Playing : 0.9366888613023362
Libraries & Demo : 0.9366888613023362
Auto & Vehicles : 0.925403453334838

In [57]:
display_table(ios_free, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


## Analysis
### IOS
- What is the most common genre? What is the runner-up?
Games
- What other patterns do you see?
the App Store is dominated by apps designed for fun
- What is the general impression — are most of the apps designed for practical purposes (education, shopping, utilities, productivity, lifestyle) or more for entertainment (games, photo and video, social networking, sports, music)?
Most of the apps for entertainment

- Can you recommend an app profile for the App Store market based on this frequency table alone? If there's a large number of apps for a particular genre, does that also imply that apps of that genre generally have a large number of users?
yes I recommend Games or entertainment in general

### Google Play
- What is the most common genre? What is the runner-up?
Family
- What other patterns do you see?
Google Play shows a more balanced landscape of both practical and fun apps.
- What is the general impression — are most of the apps designed for practical purposes (education, shopping, utilities, productivity, lifestyle) or more for entertainment (games, photo and video, social networking, sports, music)?
A little bit more for entertainment

- Can you recommend an app profile for the App Store market based on this frequency table alone? If there's a large number of apps for a particular genre, does that also imply that apps of that genre generally have a large number of users?
Tools or entertainment

## Most Popular Apps By Genre

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

### IOS - prime genre

In [58]:
prime_genre_freq_table = freq_table(ios_free, 11)
for genre in prime_genre_freq_table:
    total = 0
    len_genre = 0
    for app in ios_free:
        genre_app = app[11]
        if genre_app == genre:
            total += float(app[5])
            len_genre += 1
    avg_rating = total / len_genre
    print(genre , ' : ', avg_rating)

Social Networking  :  71548.34905660378
Photo & Video  :  28441.54375
Games  :  22788.6696905016
Music  :  57326.530303030304
Reference  :  74942.11111111111
Health & Fitness  :  23298.015384615384
Weather  :  52279.892857142855
Utilities  :  18684.456790123455
Travel  :  28243.8
Shopping  :  26919.690476190477
News  :  21248.023255813954
Navigation  :  86090.33333333333
Lifestyle  :  16485.764705882353
Entertainment  :  14029.830708661417
Food & Drink  :  33333.92307692308
Sports  :  23008.898550724636
Book  :  39758.5
Finance  :  31467.944444444445
Education  :  7003.983050847458
Productivity  :  21028.410714285714
Business  :  7491.117647058823
Catalogs  :  4004.0
Medical  :  612.0


### Google Play - category

In [61]:
category_freq_table = freq_table(android_free, 1)
for category in category_freq_table:
    total = 0
    len_category = 0
    for app in android_free:
        category_app = app[1]
        if category_app == category:
            total += float(app[5].replace(',','').replace('+',''))
            len_category += 1
    avg_rating = total / len_category
    print(category , ' : ', avg_rating)

ART_AND_DESIGN  :  1905351.6666666667
AUTO_AND_VEHICLES  :  647317.8170731707
BEAUTY  :  513151.88679245283
BOOKS_AND_REFERENCE  :  8767811.894736841
BUSINESS  :  1712290.1474201474
COMICS  :  817657.2727272727
COMMUNICATION  :  38456119.167247385
DATING  :  854028.8303030303
EDUCATION  :  3082017.543859649
ENTERTAINMENT  :  21134600.0
EVENTS  :  253542.22222222222
FINANCE  :  1387692.475609756
FOOD_AND_DRINK  :  1924897.7363636363
HEALTH_AND_FITNESS  :  4188821.9853479853
HOUSE_AND_HOME  :  1313681.9054054054
LIBRARIES_AND_DEMO  :  638503.734939759
LIFESTYLE  :  1437816.2687861272
GAME  :  15837565.085714286
FAMILY  :  2693265.4161566705
MEDICAL  :  120616.48717948717
SOCIAL  :  23253652.127118643
SHOPPING  :  7036877.311557789
PHOTOGRAPHY  :  17805627.643678162
SPORTS  :  3638640.1428571427
TRAVEL_AND_LOCAL  :  13984077.710144928
TOOLS  :  10695245.286096256
PERSONALIZATION  :  5201482.6122448975
PRODUCTIVITY  :  16787331.344927534
PARENTING  :  542603.6206896552
WEATHER  :  5074486.

In [62]:
for app in android_free:
    if app[1] == 'COMMUNICATION':
        print(app[0], ' : ', app[5])

Messenger – Text and Video Chat for Free  :  1,000,000,000+
WhatsApp Messenger  :  1,000,000,000+
Messenger for SMS  :  10,000,000+
Google Chrome: Fast & Secure  :  1,000,000,000+
Messenger Lite: Free Calls & Messages  :  100,000,000+
Gmail  :  1,000,000,000+
Hangouts  :  1,000,000,000+
Viber Messenger  :  500,000,000+
My Tele2  :  5,000,000+
Firefox Browser fast & private  :  100,000,000+
Yahoo Mail – Stay Organized  :  100,000,000+
imo beta free calls and text  :  100,000,000+
imo free video calls and chat  :  500,000,000+
Contacts  :  50,000,000+
Call Free – Free Call  :  5,000,000+
Web Browser & Explorer  :  5,000,000+
Opera Mini - fast web browser  :  100,000,000+
Browser 4G  :  10,000,000+
MegaFon Dashboard  :  10,000,000+
ZenUI Dialer & Contacts  :  10,000,000+
Cricket Visual Voicemail  :  10,000,000+
Opera Browser: Fast and Secure  :  100,000,000+
TracFone My Account  :  1,000,000+
Firefox Focus: The privacy browser  :  1,000,000+
Google Voice  :  10,000,000+
Chrome Dev  :  5,0

Eg Call  :  10,000+
ei  :  10+
EJ messenger  :  10+
Ek IRA  :  10+
Orfox: Tor Browser for Android  :  10,000,000+
EO Mumbai  :  10+
EP RSS Reader  :  100+
Voxer Walkie Talkie Messenger  :  10,000,000+
ES-1  :  500+
EU Council  :  1,000+
Council Voting Calculator  :  5,000+
Have your say on Europe  :  500+
Programi podrške EU  :  100+
Inbox.eu  :  10,000+
Everbridge  :  100,000+
Best Auto Call Recorder Free  :  500+
EZ Wifi Notification  :  10,000+
Test Server SMS FA  :  5+
Lite for Facebook Messenger  :  1,000,000+
FC Browser - Focus Privacy Browser  :  1,000+
EHiN-FH conferenceapp  :  100+
Carpooling FH Hagenberg  :  100+
Wi-Fi Auto-connect  :  1,000,000+
Talkie - Wi-Fi Calling, Chats, File Sharing  :  500,000+
WeFi - Free Fast WiFi Connect & Find Wi-Fi Map  :  1,000,000+
Sat-Fi  :  5,000+
Portable Wi-Fi hotspot Free  :  100,000+
TownWiFi | Wi-Fi Everywhere  :  500,000+
Jazz Wi-Fi  :  10,000+
Sat-Fi Voice  :  1,000+
Free Wi-fi HotspoT  :  50,000+
FN Web Radio  :  10+
FNH Payment Info 